Detectors/MUON/MCH/Raw/Encoder/Digit/digit2raw.cxx

```diff
@@ -0,0 +1,135 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "CommonDataFormat/InteractionRecord.h"
+#include "DetectorsRaw/RawFileWriter.h"
+#include "DigitEncoder.h"
+#include "DigitReader.h"
+#include "Headers/RAWDataHeader.h"
+#include "MCHBase/Digit.h"
+#include "MCHRawCommon/DataFormats.h"
+#include "MCHRawElecMap/Mapper.h"
+#include "MCHRawEncoderPayload/DataBlock.h"
+#include "MCHRawEncoderPayload/PayloadPaginator.h"
+#include <TBranch.h>
+#include <TFile.h>
+#include <TTree.h>
+#include <boost/program_options.hpp>
+#include <fmt/format.h>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <string>
+
+namespace po = boost::program_options;
+using namespace o2::mch::raw;
+
+namespace
+{
+std::string asString(const o2::InteractionRecord& ir)
+{
+  return fmt::format("ORB {:6d} BC {:4d}",
+                     ir.orbit, ir.bc);
+}
+
+std::string asString(const o2::InteractionTimeRecord& ir)
+{
+  return asString(static_cast<o2::InteractionRecord>(ir));
+}
+} // namespace
+
+std::ostream& operator<<(std::ostream& os, const o2::mch::Digit& d)
+{
+  os << fmt::format("DE {:4d} PADUID {:8d} ADC {:6d} TS {:g}",
+                    d.getDetID(), d.getPadID(), d.getADC(),
+                    d.getTimeStamp());
+  return os;
+}
+
+void digit2raw(const std::string& input,
+               DigitEncoder& encoder,
+               PayloadPaginator paginate)
```

Comment:
Any reason to pass paginate by copy rather than by ref?

Reply:
no particular reason. will change to ref (not that it matters a lot in this case though)